### PR TITLE
Blog Sample Data - New custom fields, tags

### PR DESCRIPTION
**Pull request by chmst**
### Summary of Changes
This PR adds new data to the content in blog sample data.

Tags
Fieldgroup
Field 

All loadLanguage statements come before the exception is thrown.

Modules are moved to position bottom-b

### Testing Instructions
Install blog sample data in a fresh installation.

In the backend you now must have 
a Field Group
a Field in this field group. 
four tags

In the frontend:
Check that articles have the new Tab "The Autor"
For two articles (Workflow and About your homepage) the author fields have some content.
In the frontend, the author - field must be shown below the text.

Tags are shown for articles in category blog

grafik

### Documentation Changes Required
If there is a documentation about sample data: yes

### CODE REVIEW

File: plugins/sampledata/blog/blog.php [hunk 1]
```diff
@@ -119,6 +119,108 @@ public function onAjaxSampledataApplyStep1()
...
+
+		$groupTable->title           = Text::_('PLG_SAMPLEDATA_BLOG_SAMPLEDATA_CONTENT_FIELDS_GROUP_TITLE') . $langSuffix;
+		$groupTable->description     = '';
+		$groupTable->note            = '';
+		$groupTable->state           = 1;
+		$groupTable->ordering        = 0;
+		$groupTable->created         = Factory::getDate()->toSql();
+		$groupTable->created_by      = $user->id;
+		$groupTable->modified        = Factory::getDate()->toSql();
+		$groupTable->modified_by     = $user->id;
```
Comment (richard67) [inline plugins/sampledata/blog/blog.php]:
```suggestion
		$groupTable->modified        = $groupTable->created;
		$groupTable->modified_by     = $groupTable->created_by;
```
When calling getDate 2 times, the times might not be exactly the same. But for something which never has been modified, the modification time has to be exactly the same as the creation time.

For consistency I do the same with the modified_by, even if there will not be a difference.

**Review by richard67 — CHANGES_REQUESTED** [review]
See suggested change above.

File: plugins/sampledata/blog/blog.php (hunk 1 above)
Comment (chmst) [reply plugins/sampledata/blog/blog.php]: Resolved in the new commit

**Comment by chmst** [comment]
Fixed setting the date in the field Group.

Fixed conditions for installing sample data with workflow and custom fields.

Workflow and/or custom fields can be disabled, but the articles and categories can be saved. 
Also a user could have the right to add fields but not worfklows.

**Review by richard67 — APPROVED** [review]

**Comment by richard67** [comment]
Code review good, real test has to follow.

**Comment by chmst** [comment]
@richard67 Probably it is better to revert the changes in the core fields and add them there?

**Comment by richard67** [comment]
> @richard67 Probably it is better to revert the changes in the core fields and add them there?

You mean the changes in FieldTable.php and lib_joomla.ini?

Well, that seems to be indeed better in the CMS repo. But will it be possible to test it there when not using fields in blog sample data?

**Comment by chmst** [comment]
Yes. So maybe you could make the test and then I revert changes in the field scripts?

**Comment by richard67** [comment]
Will test later tonight or tomorrow.

**Comment by chmst** [comment]
Tags are now integrated into the first step. The code would be clearer if the concerns are separated, but then we will have an issue with translation.
At the moment, there are 3 steps - more steps would require new language keys forthe  newly added steps.

**Review by richard67 — APPROVED** [review]

**Comment by richard67** [comment]
I have tested this PR :white_check_mark: with success.

**Comment by richard67** [comment]
Thanks!

**Pull request merged**